### ISSUE TITLE
Make the site data-driven

### ISSUE THREAD

**Issue by jacebenson** [caniuse-email/caniuse.email#32]
Most the enhancements are to add a email client or add a class.
This stuff should be all data driven I think so the style can managed separate then the data.

I started this a long time ago and you added it to the project as the json.htm under /docs but we should add something to make the build of the html pretty simple or have the page literally load the JSON.  

If this isn't where you want to take this project thats fine, just close this issue, but if this is something you'd like to explore I'd love to help make it a reality.

**Comment by coliff** [comment]
Hi Jace. Thanks for following up on this... I agree that we should make the sites data by handled by a separate file.

I couldn't decide if it'd be best to do this in JavaScript in the browser, or via a gulp build script which we'd run to generate a static site - or even using a static site generator like Jekyll. Overall I think I liked the idea of not needing to make a build everytime we change any of the data though so doing it in JavaScript like your proposed JSON solution might be the best idea.

The page you created is out of sync with the 'static' index.html so I need to get that updated. My JavaScript skills aren't too strong so I'll give you a shout if I need help with it. :-)  I'll try and work on this later this week.

**Comment by panchalkalpesh** [comment]
We can make use of a `data.json` the way [caniuse.com](https://github.com/Fyrd/caniuse) does. 
Another strategy is to use a feature based json, like they have it [here](https://github.com/Fyrd/caniuse/tree/master/features-json). 

Hope this helps.

**Comment by coliff** [comment]
hey! After a lot of thought and a long delay, I got around to migrating the site to Jekyll. I use static site generators quite a lot myself and I like that it outputs plain HTML (I'm not a fan of JS-frameworks). I went with Jekyll as GitHub Pages supports it out of the box. The data for the site is in the `_data` directory. There's one for clients and one for features. There is still a LOT to do. I want to add a lot of new features to track (flexbox, animation, animated gif, inline SVG) - and many more email clients. I also like the idea of having more than one page. e.g. https://caniuse.email/webmail/ which would show a table of just the webmail clients. Now the data is separate we should be able to do this with Jekyll logic.

The logic to build the tables now could do with improvements, so PRs are very welcome!
